### PR TITLE
UCT/UD: nonblocking ep_create_connected and flush fixes

### PR DESCRIPTION
**Pull request by alex-mikheev**
@yosefe 
uct_ep_create_connected will schedule sending of CREQ if it is not
possible to send CREW immediately.

flush will return IN_PROGRESS until outstanding CREQ/user
pending requests are completed

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1012/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1625/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1017/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1018/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1631/ for details.

File: src/uct/ib/ud/base/ud_ep.c [hunk 1]
```diff
@@ -507,7 +505,7 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
         }
         ucs_trace_data("DUP/OOB - schedule ack, head_sn=%d sn=%d", 
                        ep->rx.ooo_pkts.head_sn, neth->psn);
-        uct_ud_ep_ctl_op_add(iface, ep, UCT_UD_EP_OP_ACK);
+        uct_ud_ep_ctl_op_add(iface, ep, UCT_UD_EP_OP_ACK);
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_ep.c]: added space in end of line?

File: test/gtest/uct/test_ud_pending.cc [hunk 2]
```diff
@@ -43,6 +43,33 @@ class test_ud_pending : public ud_base_test {
         return UCS_ERR_BUSY;
     }
 
+    uct_pending_req_t r[64];
```
Comment (yosefe) [inline test/gtest/uct/test_ud_pending.cc]: can this be a local variable?

Comment (alex-mikheev) [inline test/gtest/uct/test_ud_pending.cc]: no. requests are posted in post_pending_reqs() and are completed during progress or flush. So i can not have them on stack.

Comment (yosefe) [inline test/gtest/uct/test_ud_pending.cc]: i see. so can it be a member variable?

Comment (alex-mikheev) [inline test/gtest/uct/test_ud_pending.cc]: it is already a member. Do you want me to rename it to m_r ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1640/ for details.

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1029/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1643/ for details.

**Comment by yosefe** [comment]
:+1:

**Pull request merged**